src/language-js/embed/html.js

```diff
@@ -12,15 +12,50 @@ import {
 } from "../print/template-literal.js";
 import { isAngularComponentTemplate, hasLanguageComment } from "./utils.js";
 
+function getVariableName(expression) {
+  if (expression.type === "Identifier") {
+    return expression.name;
+  }
+
+  if (expression.type === "ThisExpression") {
+    return "this";
+  }
+
+  if (expression.type === "MemberExpression") {
```

Comment:
Let's ignore computed and optional(just ti be safe)

**Prettier pr-15545**
[Playground link](https://deploy-preview-15545--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEALGBbANgAwDpQAEhAPACTACGAdAB4C+AfCQPQWUDatAukwflBAAaEBAAOMAJbQAzslCUAToogB3AApKEclJSyrKATzkiARospgA1nBgBlShjgAZSVDjIAZnplwzF61s7MUs3AHNkGEUAVz8QXwxJSJi4mXCsOABFaIh4Lx84gCsZWjt0rJy8pG8sXxEAR0q4dRUxHRBKGQBadzgAE37hECjKSSxwgGEIDAxKZA6sLCG0qDCMgEEYKMlTaPh1OEVXd3zauPRsAHVUSXgZELA4O21byQA3W8N5sBkTEDfYgBJKADWB2MCKSQSdYguwwQwZU51EBiFS+S4WMTzVFwXyKN4eERuPEwFqUMKzJFxEKKPHzUyUUxwJYiVFuGCXSR9GCoZAADgADCJFHBGpIRWSKXNqgURDBGZzubykAAmETRXwAFUZOhqyLgGCZfQGfWclFW0XJcAAYhBFLMtuF5pQ9hAQPR6EA)
<!-- prettier-ignore -->
```sh
--parser babel
```

**Input:**
<!-- prettier-ignore -->
```jsx
html`
  <${a.x}></${a[x]}>
`

```

**Output:**
<!-- prettier-ignore -->
```jsx
html` <${a.x}></${a.x}> `;

```